homeassistant/components/freebox/alarm_control_panel.py

```diff
@@ -76,63 +72,55 @@ def __init__(
+        self._set_features()
 
     async def async_alarm_disarm(self, code: str | None = None) -> None:
         """Send disarm command."""
-        if await self.set_home_endpoint_value(self._command_disarm):
-            self._set_state(STATE_ALARM_DISARMED)
+        await self.set_home_endpoint_value(self._command_disarm)
 
     async def async_alarm_arm_away(self, code: str | None = None) -> None:
         """Send arm away command."""
-        if await self.set_home_endpoint_value(self._command_arm_away):
-            self._set_state(STATE_ALARM_ARMING)
+        await self.set_home_endpoint_value(self._command_arm_away)
 
     async def async_alarm_arm_home(self, code: str | None = None) -> None:
         """Send arm home command."""
-        if await self.set_home_endpoint_value(self._command_arm_home):
-            self._set_state(STATE_ALARM_ARMING)
+        await self.set_home_endpoint_value(self._command_arm_home)
 
     async def async_alarm_trigger(self, code: str | None = None) -> None:
         """Send alarm trigger command."""
-        if await self.set_home_endpoint_value(self._command_trigger):
-            self._set_state(STATE_ALARM_TRIGGERED)
+        await self.set_home_endpoint_value(self._command_trigger)
 
-    async def async_update_signal(self):
-        """Update signal."""
-        state = await self.get_home_endpoint_value(self._command_state)
+    async def async_update(self) -> None:
+        """Update state."""
+        state: str | None = await self.get_home_endpoint_value(self._command_state)
         if state:
-            self._set_state(state)
+            self._attr_state = FREEBOX_TO_STATUS.get(state)
+        else:
+            self._attr_state = STATE_ALARM_DISARMED
 
-    def _set_features(self, node: dict[str, Any]) -> None:
-        """Add alarm features."""
+    def _set_features(self) -> None:
+        """Set alarm features."""
         # Search if the arm home feature is present => has an "alarm2" endpoint
-        can_arm_home = False
-        for nodeid, local_node in self._router.home_devices.items():
-            if nodeid == local_node["id"]:
-                alarm2 = next(
-                    filter(
-                        lambda x: (x["name"] == "alarm2" and x["ep_type"] == "signal"),
-                        local_node["show_endpoints"],
-                    ),
-                    None,
-                )
-                if alarm2:
-                    can_arm_home = alarm2["value"]
-                    break
-
-        if can_arm_home:
+        can_arm_home = next(
+            (
+                endpoint
+                for endpoint in self._node["type"]["endpoints"]
+                if endpoint["name"] == "alarm2" and endpoint["ep_type"] == "signal"
+            ),
+            None,
+        )
+
+        if (
+            self._command_arm_home and can_arm_home
+        ):  # or check self.get_value("slot", "alarm2") ???
```

Comment:
As written here https://github.com/home-assistant/core/pull/103389/files#r1393219407

Based on test fixtures, to check the arm home feature either:
1. go check to `node["show_endpoints"]` -> `"name": "alarm2"` + `"ep_type": "slot"` (yes, **slot**) -> check `"value": true`
    - `self.get_value("slot", "alarm2")` solution
2. go check `node["type"]["endpoints"]` (yes, **["type"]["endpoints"]**)  -> `"name": "alarm2"` + `"ep_type": "signal"` presence
    - `can_arm_home` solution

Otherwise, it's not working

@nachonam :
Do you have an alarm with arm home feature ?
What is your advise: solution 1 with `get_value` or solution 2 with `can_arm_home` ?

Is alarm test fixtures up to date ? (it seems, but just in case)

Reply:
`        # Search if the arm home feature is present => has an "alarm2" endpoint
        can_arm_home = next(
            (
                endpoint
                for endpoint in self._node["type"]["endpoints"]
                if endpoint["name"] == "alarm2" and endpoint["ep_type"] == "slot"
            ),
            None,
        )
        # if feature is found, can_arm_home is not None
        if (
            self._command_arm_home and can_arm_home is not None
        ):  # or check self.get_value("slot", "alarm2") ???
            self._attr_supported_features = (
                AlarmControlPanelEntityFeature.ARM_AWAY
                | AlarmControlPanelEntityFeature.ARM_HOME
                | AlarmControlPanelEntityFeature.TRIGGER
            )`

Reply:
So you are saying `["type"]["endpoints"] with "slot"`.

As I remember (not having the code under my eyes now), this test fixture exists but `["type"]["endpoints"] with "signal"` and `["show_endpoints"] with "slot"` are not present.

Compared to other Freebox Home devices that when they got `alarm2` they got all 3 fixtures, check PIR and DWS.

---

`if can_arm_home` already check not None, not 0 and other falsy values.

Reply:
Debugging:
![image](https://github.com/home-assistant/core/assets/39791/df4e8477-1aae-4239-b22e-32fd67dc5781)
 
![image](https://github.com/home-assistant/core/assets/39791/ab5a1445-a458-4305-b100-d2ae77e095ac)

Reply:
So, `can_arm_home` == `self._command_arm_home` then